src/order/filter/extr.lean

```diff
@@ -477,3 +477,54 @@ lemma is_max_on.max (hf : is_max_on f s a) (hg : is_max_on g s a) :
 hf.max hg
 
 end decidable_linear_order
+
+section eventually
+
+/-! ### Relation with `eventually` comparisons of two functions -/
+
+lemma filter.eventually_le.is_max_filter {α β : Type*} [preorder β] {f g : α → β} {a : α} {l : filter α}
+  (heq : g ≤ᶠ[l] f) (hfga : f a = g a) (h : is_max_filter f l a) :
+  is_max_filter g l a :=
+begin
+  simp [is_max_filter, hfga, eventually_le, eventually_iff_exists_mem] at *,
```

Comment:
`simp` should be used only as a finishing tactic because it's not very predictable (e.g., what if new `simp` lemmas will simplify the goal more than you expected?).

Actually, this proof doesn't require you using this `simp` at all. You can start with `refine heq.mp (h.mono $ λ x hf hgf, _)`. BTW, please rename `heq` to `hle`.